metric_groups does not work with the -all option

**Describe the bug**
The following config:

```yml
table_name: customer
metric_groups:
    - all
```
Throws this exception:
```
soda scan warehouse.yml tables/metric_groups.yml
  | 
  | Scanning tables/metric_groups.yml ...
  | Invalid key in tables/metric_groups.yml.metrics : i
  | Invalid key in tables/metric_groups.yml.metrics : l
  | Invalid key in tables/metric_groups.yml.metrics : o
  | Invalid key in tables/metric_groups.yml.metrics : g
  | Invalid key in tables/metric_groups.yml.metrics : u
  | Invalid key in tables/metric_groups.yml.metrics : d
  | Invalid key in tables/metric_groups.yml.metrics : t
  | Invalid key in tables/metric_groups.yml.metrics : n
  | Invalid key in tables/metric_groups.yml.metrics : h
  | Invalid key in tables/metric_groups.yml.metrics : v
  | Invalid key in tables/metric_groups.yml.metrics : y
  | Invalid key in tables/metric_groups.yml.metrics : f
  | Invalid key in tables/metric_groups.yml.metrics : s
  | Invalid key in tables/metric_groups.yml.metrics : m
  | Invalid key in tables/metric_groups.yml.metrics : a
  | Invalid key in tables/metric_groups.yml.metrics : c
  | Invalid key in tables/metric_groups.yml.metrics : e
  | Invalid key in tables/metric_groups.yml.metrics : p
  | Invalid key in tables/metric_groups.yml.metrics : r
  | Scan failed: tables/metric_groups.yml configuration errors: 
  [warning] (i) Invalid key in tables/metric_groups.yml.metrics : i
  [warning] (l) Invalid key in tables/metric_groups.yml.metrics : l
  [warning] (o) Invalid key in tables/metric_groups.yml.metrics : o
  [warning] (g) Invalid key in tables/metric_groups.yml.metrics : g
  [warning] (u) Invalid key in tables/metric_groups.yml.metrics : u
  [warning] (d) Invalid key in tables/metric_groups.yml.metrics : d
  [warning] (t) Invalid key in tables/metric_groups.yml.metrics : t
  [warning] (n) Invalid key in tables/metric_groups.yml.metrics : n
  [warning] (h) Invalid key in tables/metric_groups.yml.metrics : h
  [warning] (v) Invalid key in tables/metric_groups.yml.metrics : v
  [warning] (y) Invalid key in tables/metric_groups.yml.metrics : y
  [warning] (f) Invalid key in tables/metric_groups.yml.metrics : f
  [warning] (s) Invalid key in tables/metric_groups.yml.metrics : s
  [warning] (m) Invalid key in tables/metric_groups.yml.metrics : m
  [warning] (a) Invalid key in tables/metric_groups.yml.metrics : a
  [warning] (c) Invalid key in tables/metric_groups.yml.metrics : c
  [warning] (e) Invalid key in tables/metric_groups.yml.metrics : e
  [warning] (p) Invalid key in tables/metric_groups.yml.metrics : p
  [warning] (r) Invalid key in tables/metric_groups.yml.metrics : r
Traceback (most recent call last):
  File "/Users/albin/dev/py/soda-sql/sodasql/cli/cli.py", line 366, in scan
    scan = scan_builder.build()
  File "/Users/albin/dev/py/soda-sql/sodasql/scan/scan_builder.py", line 76, in build
    parser.assert_no_warnings_or_errors()
  File "/Users/albin/dev/py/soda-sql/sodasql/scan/parser.py", line 120, in assert_no_warnings_or_errors
    raise AssertionError(f'{self.description} configuration errors: \n  '
AssertionError: tables/metric_groups.yml configuration errors: 
  [warning] (i) Invalid key in tables/metric_groups.yml.metrics : i
  [warning] (l) Invalid key in tables/metric_groups.yml.metrics : l
  [warning] (o) Invalid key in tables/metric_groups.yml.metrics : o
  [warning] (g) Invalid key in tables/metric_groups.yml.metrics : g
  [warning] (u) Invalid key in tables/metric_groups.yml.metrics : u
  [warning] (d) Invalid key in tables/metric_groups.yml.metrics : d
  [warning] (t) Invalid key in tables/metric_groups.yml.metrics : t
  [warning] (n) Invalid key in tables/metric_groups.yml.metrics : n
  [warning] (h) Invalid key in tables/metric_groups.yml.metrics : h
  [warning] (v) Invalid key in tables/metric_groups.yml.metrics : v
  [warning] (y) Invalid key in tables/metric_groups.yml.metrics : y
  [warning] (f) Invalid key in tables/metric_groups.yml.metrics : f
  [warning] (s) Invalid key in tables/metric_groups.yml.metrics : s
  [warning] (m) Invalid key in tables/metric_groups.yml.metrics : m
  [warning] (a) Invalid key in tables/metric_groups.yml.metrics : a
  [warning] (c) Invalid key in tables/metric_groups.yml.metrics : c
  [warning] (e) Invalid key in tables/metric_groups.yml.metrics : e
  [warning] (p) Invalid key in tables/metric_groups.yml.metrics : p
  [warning] (r) Invalid key in tables/metric_groups.yml.metrics : r
  | Exiting with code 1
```

**To Reproduce**
Steps to reproduce the behavior:
1. Run a scan with the above config


**Context**
__Include your scan.yml or warehouse.yml when relevant__

**OS**: Mac OS 11.2.3
**Python Version**: 3.9.2
**Soda SQL Version**: 2.1.0b1
**Warehouse Type**: postgres

--- a/core/sodasql/scan/metric.py
+++ b/core/sodasql/scan/metric.py
@@ -11,81 +11,81 @@
 
 
 class Metric:
-
     ROW_COUNT = 'row_count'
     SCHEMA = 'schema'
 
-    MISSING_COUNT = 'missing_count'
-    MISSING_PERCENTAGE = 'missing_percentage'
-    VALUES_COUNT = 'values_count'
-    VALUES_PERCENTAGE = 'values_percentage'
-    VALID_COUNT = 'valid_count'
-    VALID_PERCENTAGE = 'valid_percentage'
-    INVALID_COUNT = 'invalid_count'
-    INVALID_PERCENTAGE = 'invalid_percentage'
-    MIN = 'min'
-    MAX = 'max'
     AVG = 'avg'
-    SUM = 'sum'
-    VARIANCE = 'variance'
-    STDDEV = 'stddev'
-    MIN_LENGTH = 'min_length'
-    MAX_LENGTH = 'max_length'
     AVG_LENGTH = 'avg_length'
     DISTINCT = 'distinct'
-    UNIQUE_COUNT = 'unique_count'
     DUPLICATE_COUNT = 'duplicate_count'
-    UNIQUENESS = 'uniqueness'
-    MAXS = 'maxs'
-    MINS = 'mins'
     FREQUENT_VALUES = 'frequent_values'
     HISTOGRAM = 'histogram'
-
-    METRIC_GROUP_MISSING = 'missing'
-    METRIC_GROUP_VALIDITY = 'validity'
-    METRIC_GROUP_DUPLICATES = 'duplicates'
-    METRIC_GROUP_STATISTICS = 'statistics'
-    METRIC_GROUP_LENGTH = 'length'
-    METRIC_GROUP_PROFILING = 'profiling'
-    METRIC_GROUP_ALL = 'all'
-
-    METRIC_GROUPS = {
-        METRIC_GROUP_MISSING: [MISSING_COUNT, MISSING_PERCENTAGE, VALUES_COUNT, VALUES_PERCENTAGE],
-        METRIC_GROUP_VALIDITY: [VALID_COUNT, VALID_PERCENTAGE, INVALID_COUNT, INVALID_PERCENTAGE],
-        METRIC_GROUP_DUPLICATES: [DISTINCT, UNIQUE_COUNT, UNIQUENESS, DUPLICATE_COUNT],
-        METRIC_GROUP_STATISTICS: [MIN, MAX, AVG, SUM, VARIANCE, STDDEV],
-        METRIC_GROUP_LENGTH: [MIN_LENGTH, MAX_LENGTH, AVG_LENGTH],
-        METRIC_GROUP_PROFILING: [MAXS, MINS, FREQUENT_VALUES, HISTOGRAM],
-        METRIC_GROUP_ALL: (METRIC_GROUP_MISSING + METRIC_GROUP_VALIDITY + METRIC_GROUP_DUPLICATES +
-                           METRIC_GROUP_STATISTICS + METRIC_GROUP_LENGTH + METRIC_GROUP_PROFILING)
-    }
+    INVALID_COUNT = 'invalid_count'
+    INVALID_PERCENTAGE = 'invalid_percentage'
+    MAX = 'max'
+    MAXS = 'maxs'
+    MAX_LENGTH = 'max_length'
+    MIN = 'min'
+    MINS = 'mins'
+    MIN_LENGTH = 'min_length'
+    MISSING_COUNT = 'missing_count'
+    MISSING_PERCENTAGE = 'missing_percentage'
+    STDDEV = 'stddev'
+    SUM = 'sum'
+    UNIQUENESS = 'uniqueness'
+    UNIQUE_COUNT = 'unique_count'
+    VALID_COUNT = 'valid_count'
+    VALID_PERCENTAGE = 'valid_percentage'
+    VALUES_COUNT = 'values_count'
+    VALUES_PERCENTAGE = 'values_percentage'
+    VARIANCE = 'variance'
 
     METRIC_TYPES = [
         ROW_COUNT,
         SCHEMA,
-        MISSING_COUNT,
-        MISSING_PERCENTAGE,
-        VALUES_COUNT,
-        VALUES_PERCENTAGE,
-        VALID_COUNT,
-        VALID_PERCENTAGE,
-        INVALID_COUNT,
-        INVALID_PERCENTAGE,
-        MIN,
-        MAX,
+
         AVG,
-        SUM,
-        VARIANCE,
-        STDDEV,
-        MIN_LENGTH,
-        MAX_LENGTH,
         AVG_LENGTH,
         DISTINCT,
-        UNIQUE_COUNT,
-        UNIQUENESS,
         DUPLICATE_COUNT,
-        MAXS,
-        MINS,
         FREQUENT_VALUES,
-        HISTOGRAM
+        HISTOGRAM,
+        INVALID_COUNT,
+        INVALID_PERCENTAGE,
+        MAX,
+        MAXS,
+        MAX_LENGTH,
+        MIN,
+        MINS,
+        MIN_LENGTH,
+        MISSING_COUNT,
+        MISSING_PERCENTAGE,
+        STDDEV,
+        SUM,
+        UNIQUENESS,
+        UNIQUE_COUNT,
+        VALID_COUNT,
+        VALID_PERCENTAGE,
+        VALUES_COUNT,
+        VALUES_PERCENTAGE,
+        VARIANCE,
     ]
+
+    METRIC_GROUP_ALL = 'all'
+    METRIC_GROUP_DUPLICATES = 'duplicates'
+    METRIC_GROUP_LENGTH = 'length'
+    METRIC_GROUP_MISSING = 'missing'
+    METRIC_GROUP_PROFILING = 'profiling'
+    METRIC_GROUP_STATISTICS = 'statistics'
+    METRIC_GROUP_VALIDITY = 'validity'
+
+    METRIC_GROUPS = {
+        METRIC_GROUP_DUPLICATES: [DISTINCT, DUPLICATE_COUNT, UNIQUENESS, UNIQUE_COUNT],
+        METRIC_GROUP_LENGTH: [AVG_LENGTH, MAX_LENGTH, MIN_LENGTH],
+        METRIC_GROUP_MISSING: [MISSING_COUNT, MISSING_PERCENTAGE, VALUES_COUNT, VALUES_PERCENTAGE],
+        METRIC_GROUP_PROFILING: [FREQUENT_VALUES, HISTOGRAM, MAXS, MINS],
+        METRIC_GROUP_STATISTICS: [AVG, SUM, MAX, MIN, STDDEV, VARIANCE],
+        METRIC_GROUP_VALIDITY: [INVALID_COUNT, INVALID_PERCENTAGE, VALID_COUNT, VALID_PERCENTAGE],
+    }
+
+    # METRIC_GROUPS[METRIC_GROUP_ALL] = list(itertools.chain.from_iterable(METRIC_GROUPS.values()))

--- a/core/sodasql/scan/scan.py
+++ b/core/sodasql/scan/scan.py
@@ -187,6 +187,10 @@ class Scan:
                         if scan_column.non_missing_and_valid_condition \
                         else dialect.sql_expr_length(scan_column.qualified_column_name)
 
+                    if self.scan_yml.is_metric_enabled(Metric.AVG_LENGTH, column_name):
+                        fields.append(dialect.sql_expr_avg(length_expr))
+                        measurements.append(Measurement(Metric.AVG_LENGTH, column_name))
+
                     if self.scan_yml.is_metric_enabled(Metric.MIN_LENGTH, column_name):
                         fields.append(dialect.sql_expr_min(length_expr))
                         measurements.append(Measurement(Metric.MIN_LENGTH, column_name))

--- a/core/sodasql/scan/scan_column.py
+++ b/core/sodasql/scan/scan_column.py
@@ -67,7 +67,7 @@ class ScanColumn:
 
             self.validity_format = self.scan_yml.get_validity_format(column_metadata)
             self.is_valid_enabled = \
-                (self.validity is not None and self.is_validity_metric_enabled) \
+                (self.validity is not None or self.is_validity_metric_enabled) \
                 or self.scan_yml.is_any_metric_enabled([Metric.DISTINCT, Metric.UNIQUENESS], self.column_name)
 
             self.is_missing_enabled = self.is_valid_enabled or self.is_missing_metric_enabled

--- a/core/sodasql/scan/scan_yml_parser.py
+++ b/core/sodasql/scan/scan_yml_parser.py
@@ -147,28 +147,32 @@ class ScanYmlParser(Parser):
 
         self._push_context(metrics, KEY_METRICS)
 
-        for metric_group_name in Metric.METRIC_GROUPS:
-            group_metrics = Metric.METRIC_GROUPS[metric_group_name]
-            for metric in metrics:
-                if metric in group_metrics:
-                    metrics_groups.add(metric_group_name)
+        # Add special case for all
+        if Metric.METRIC_GROUP_ALL in metrics_groups:
+            metrics.update(Metric.METRIC_TYPES)
+        else:
+            for metric_group_name in Metric.METRIC_GROUPS:
+                group_metrics = Metric.METRIC_GROUPS[metric_group_name]
+                for metric in metrics:
+                    if metric in group_metrics:
+                        metrics_groups.add(metric_group_name)
 
-        if Metric.METRIC_GROUP_VALIDITY in metrics_groups:
-            metrics_groups.add(Metric.METRIC_GROUP_MISSING)
+            if Metric.METRIC_GROUP_VALIDITY in metrics_groups:
+                metrics_groups.add(Metric.METRIC_GROUP_MISSING)
 
-        if Metric.METRIC_GROUP_MISSING in metrics_groups:
-            metrics.add(Metric.ROW_COUNT)
+            if Metric.METRIC_GROUP_MISSING in metrics_groups:
+                metrics.add(Metric.ROW_COUNT)
 
-        for metric_group_name in metrics_groups:
-            if metric_group_name in Metric.METRIC_GROUPS:
-                for group_metric in Metric.METRIC_GROUPS[metric_group_name]:
-                    metrics.add(group_metric)
-            else:
-                self.warning(f'Invalid metric_group {metric_group_name}')
+            for metric_group_name in metrics_groups:
+                if metric_group_name in Metric.METRIC_GROUPS:
+                    for group_metric in Metric.METRIC_GROUPS[metric_group_name]:
+                        metrics.add(group_metric)
+                else:
+                    self.warning(f'Invalid metric_group {metric_group_name}')
 
-        if Metric.HISTOGRAM in metrics:
-            metrics.add(Metric.MIN)
-            metrics.add(Metric.MAX)
+            if Metric.HISTOGRAM in metrics:
+                metrics.add(Metric.MIN)
+                metrics.add(Metric.MAX)
 
         self.check_invalid_keys(Metric.METRIC_TYPES)
 
@@ -393,8 +397,10 @@ class ScanYmlParser(Parser):
 
         test_name = sql_metric_name
         test_expression = f'{test_name} == 0'
-        test_id = self.create_test_id(test_expression, test_name, sql_metric_index, column_name, sql_metric_name, sql_metric_index)
-        test_title = self.create_test_title(test_expression, test_name, sql_metric_index, column_name, sql_metric_name, sql_metric_index)
+        test_id = self.create_test_id(test_expression, test_name, sql_metric_index, column_name, sql_metric_name,
+                                      sql_metric_index)
+        test_title = self.create_test_title(test_expression, test_name, sql_metric_index, column_name, sql_metric_name,
+                                            sql_metric_index)
 
         test = Test(id=test_id,
                     title=test_title,

--- a/tests/common/sql_test_case.py
+++ b/tests/common/sql_test_case.py
@@ -50,7 +50,6 @@ def equals_ignore_case(left, right):
 
 
 class SqlTestCase(TestCase):
-
     warehouse_fixtures_by_target = {}
     warehouses_close_enabled = True
 
@@ -184,7 +183,7 @@ class SqlTestCase(TestCase):
             fields.append(f'SUM({qualified_column_name})')
 
         sql = 'SELECT \n  ' + ',\n  '.join(fields) + ' \n' \
-              'FROM ' + scan.qualified_table_name
+                                                     'FROM ' + scan.qualified_table_name
 
         where_clauses = []
 

--- a/tests/local/warehouse/metrics/test_all_table_metrics.py
+++ b/tests/local/warehouse/metrics/test_all_table_metrics.py
@@ -1,0 +1,305 @@
+#  Copyright 2020 Soda
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#   http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from sodasql.scan.metric import Metric
+from sodasql.scan.scan_yml_parser import KEY_METRIC_GROUPS
+from tests.common.sql_test_case import SqlTestCase
+from decimal import *
+
+
+class TestMetricGroups(SqlTestCase):
+
+    def test_metric_group_duplicates(self):
+        self.sql_recreate_table(
+            [f"score {self.dialect.data_type_varchar_255}"],
+            ["('1')",
+             "('2')",
+             "('2')",
+             "('3')",
+             "('3')",
+             "('3')",
+             "('3')",
+             "('3')",
+             "('4')",
+             "('4')",
+             "('5')",
+             "(null)"])
+
+        scan_result = self.scan({
+            KEY_METRIC_GROUPS: [
+                Metric.METRIC_GROUP_DUPLICATES
+            ]
+        })
+
+        self.assertEqual(scan_result.get(Metric.DISTINCT), 5)
+        self.assertEqual(scan_result.get(Metric.DUPLICATE_COUNT), 3)
+        self.assertEqual(scan_result.get(Metric.UNIQUENESS), 40)
+        self.assertEqual(scan_result.get(Metric.UNIQUE_COUNT), 2)
+
+    def test_metric_group_length(self):
+        self.sql_recreate_table(
+            [f"score {self.dialect.data_type_varchar_255}"],
+            ["('1')",
+             "('2')",
+             "('2')",
+             "('3')",
+             "('3')",
+             "('3')",
+             "('3')",
+             "('3')",
+             "('4')",
+             "('4')",
+             "('5')",
+             "(null)"])
+
+        scan_result = self.scan({
+            KEY_METRIC_GROUPS: [
+                Metric.METRIC_GROUP_LENGTH
+            ]
+        })
+        self.assertEqual(scan_result.get(Metric.AVG_LENGTH), 1)
+        self.assertEqual(scan_result.get(Metric.MAX_LENGTH), 1)
+        self.assertEqual(scan_result.get(Metric.MIN_LENGTH), 1)
+
+    def test_metric_group_missing(self):
+        self.sql_recreate_table(
+            [f"score {self.dialect.data_type_varchar_255}"],
+            ["('1')",
+             "('2')",
+             "('2')",
+             "('3')",
+             "('3')",
+             "('3')",
+             "('3')",
+             "('3')",
+             "('4')",
+             "('4')",
+             "('5')",
+             "(null)"])
+
+        scan_result = self.scan({
+            KEY_METRIC_GROUPS: [
+                Metric.METRIC_GROUP_MISSING
+            ]
+        })
+        self.assertEqual(scan_result.get(Metric.MISSING_COUNT), 1)
+        self.assertEqual(scan_result.get(Metric.MISSING_PERCENTAGE), 8.333333333333334)  # (1/12)%
+        self.assertEqual(scan_result.get(Metric.VALUES_COUNT), 11)
+        self.assertEqual(scan_result.get(Metric.VALUES_PERCENTAGE), 91.66666666666667)  # (11/12)%
+
+    def test_metric_group_profiling(self):
+        self.sql_recreate_table(
+            [f"score {self.dialect.data_type_varchar_255}"],
+            ["('1')",
+             "('2')",
+             "('2')",
+             "('3')",
+             "('3')",
+             "('3')",
+             "('3')",
+             "('3')",
+             "('4')",
+             "('4')",
+             "('5')",
+             "(null)"])
+
+        scan_result = self.scan({
+            KEY_METRIC_GROUPS: [
+                Metric.METRIC_GROUP_PROFILING
+            ]
+        })
+        self.assertCountEqual(scan_result.get(Metric.FREQUENT_VALUES),
+                              [{'frequency': 5, 'value': '3'},
+                               {'frequency': 2, 'value': '2'},
+                               {'frequency': 2, 'value': '4'},
+                               {'frequency': 1, 'value': '5'},
+                               {'frequency': 1, 'value': '1'}]
+                              )
+
+        self.assertEqual(scan_result.get(Metric.MAXS), ['5', '4', '3', '2', '1'])
+        self.assertEqual(scan_result.get(Metric.MINS), ['1', '2', '3', '4', '5'])
+
+    def test_metric_group_statistics(self):
+        self.sql_recreate_table(
+            [f"score {self.dialect.data_type_integer}"],
+            ["(1)",
+             "(2)",
+             "(2)",
+             "(3)",
+             "(3)",
+             "(3)",
+             "(3)",
+             "(3)",
+             "(4)",
+             "(4)",
+             "(5)",
+             "(null)"])
+
+        scan_result = self.scan({
+            KEY_METRIC_GROUPS: [
+                Metric.METRIC_GROUP_STATISTICS
+            ]
+        })
+
+        self.assertEqual(scan_result.get(Metric.AVG), 3)
+        self.assertEqual(scan_result.get(Metric.SUM), 33)
+        self.assertEqual(scan_result.get(Metric.MAX), 5)
+        self.assertEqual(scan_result.get(Metric.MIN), 1)
+        self.assertAlmostEqual(scan_result.get(Metric.STDDEV), Decimal(1.09544), 4)
+        self.assertAlmostEqual(scan_result.get(Metric.VARIANCE), Decimal(1.2000), 4)
+
+    def test_metric_group_validity(self):
+        self.sql_recreate_table(
+            [f"score {self.dialect.data_type_varchar_255}"],
+            ["('1')",
+             "('2')",
+             "('2')",
+             "('3')",
+             "('3')",
+             "('3')",
+             "('3')",
+             "('3')",
+             "('4')",
+             "('4')",
+             "('5')",
+             "(null)"])
+
+        scan_result = self.scan({
+            KEY_METRIC_GROUPS: [
+                Metric.METRIC_GROUP_VALIDITY
+            ]
+        })
+
+        self.assertEqual(scan_result.get(Metric.INVALID_COUNT), 0)
+        self.assertEqual(scan_result.get(Metric.INVALID_PERCENTAGE), 0.0)
+        self.assertEqual(scan_result.get(Metric.VALID_COUNT), 11)
+        self.assertEqual(scan_result.get(Metric.VALID_PERCENTAGE), 91.66666666666667)
+
+    def test_metric_group_all_text(self):
+        self.sql_recreate_table(
+            [f"score {self.dialect.data_type_varchar_255}"],
+            ["(1)",
+             "(2)",
+             "(2)",
+             "(3)",
+             "(3)",
+             "(3)",
+             "(3)",
+             "(3)",
+             "(4)",
+             "(4)",
+             "(5)",
+             "(null)"])
+
+        scan_result = self.scan({
+            KEY_METRIC_GROUPS: [
+                Metric.METRIC_GROUP_ALL
+            ]
+        })
+        self.assertEqual(scan_result.get(Metric.ROW_COUNT), 12)
+        self.assertEqual(scan_result.get(Metric.SCHEMA), [{'name': 'score', 'type': 'character varying'}])
+        self.assertEqual(scan_result.get(Metric.AVG_LENGTH), 1)
+        self.assertEqual(scan_result.get(Metric.DISTINCT), 5)
+        self.assertEqual(scan_result.get(Metric.DUPLICATE_COUNT), 3)
+        self.assertCountEqual(scan_result.get(Metric.FREQUENT_VALUES), [{'frequency': 5, 'value': '3'},
+                                                                        {'frequency': 2, 'value': '2'},
+                                                                        {'frequency': 2, 'value': '4'},
+                                                                        {'frequency': 1, 'value': '5'},
+                                                                        {'frequency': 1, 'value': '1'}])
+        self.assertEqual(scan_result.get(Metric.INVALID_COUNT), 0)
+        self.assertEqual(scan_result.get(Metric.INVALID_PERCENTAGE), 0)
+        self.assertEqual(scan_result.get(Metric.MAXS), ['5', '4', '3', '2', '1'])
+        self.assertEqual(scan_result.get(Metric.MAX_LENGTH), 1)
+        self.assertEqual(scan_result.get(Metric.MINS), ['1', '2', '3', '4', '5'])
+        self.assertEqual(scan_result.get(Metric.MIN_LENGTH), 1)
+        self.assertEqual(scan_result.get(Metric.MISSING_COUNT), 1)
+        self.assertEqual(scan_result.get(Metric.MISSING_PERCENTAGE), 8.333333333333334)
+        self.assertEqual(scan_result.get(Metric.UNIQUENESS), 40)
+        self.assertEqual(scan_result.get(Metric.UNIQUE_COUNT), 2)
+        self.assertEqual(scan_result.get(Metric.VALID_COUNT), 11)
+        self.assertEqual(scan_result.get(Metric.VALID_PERCENTAGE), 91.66666666666667)
+        self.assertEqual(scan_result.get(Metric.VALUES_COUNT), 11)
+        self.assertEqual(scan_result.get(Metric.VALUES_PERCENTAGE), 91.66666666666667)
+        self.assertEqual(20, len(scan_result.measurements))
+
+    def test_metric_group_all_number(self):
+        self.sql_recreate_table(
+            [f"score {self.dialect.data_type_integer}"],
+            ["(1)",
+             "(2)",
+             "(2)",
+             "(3)",
+             "(3)",
+             "(3)",
+             "(3)",
+             "(3)",
+             "(4)",
+             "(4)",
+             "(5)",
+             "(null)"])
+
+        scan_result = self.scan({
+            KEY_METRIC_GROUPS: [
+                Metric.METRIC_GROUP_ALL
+            ]
+        })
+
+        self.assertEqual(scan_result.get(Metric.ROW_COUNT), 12)
+        self.assertEqual(scan_result.get(Metric.SCHEMA), [{'name': 'score', 'type': 'integer'}])
+        self.assertEqual(scan_result.get(Metric.AVG), 3.0)
+        self.assertEqual(scan_result.get(Metric.DISTINCT), 5)
+        self.assertEqual(scan_result.get(Metric.DUPLICATE_COUNT), 3)
+        self.assertCountEqual(scan_result.get(Metric.FREQUENT_VALUES), [{'frequency': 5, 'value': 3},
+                                                                        {'frequency': 2, 'value': 2},
+                                                                        {'frequency': 2, 'value': 4},
+                                                                        {'frequency': 1, 'value': 5},
+                                                                        {'frequency': 1, 'value': 1}])
+        self.assertDictEqual(scan_result.get(Metric.HISTOGRAM), {'boundaries': [1.0,
+                                                                                1.2,
+                                                                                1.4,
+                                                                                1.6,
+                                                                                1.8,
+                                                                                2.0,
+                                                                                2.2,
+                                                                                2.4,
+                                                                                2.6,
+                                                                                2.8,
+                                                                                3.0,
+                                                                                3.2,
+                                                                                3.4,
+                                                                                3.6,
+                                                                                3.8,
+                                                                                4.0,
+                                                                                4.2,
+                                                                                4.4,
+                                                                                4.6,
+                                                                                4.8,
+                                                                                5.0],
+                                                                 'frequencies': [1, 0, 0, 0, 0, 2, 0, 0, 0, 0, 5, 0, 0,
+                                                                                 0, 0, 2, 0, 0, 0, 1]})
+        self.assertEqual(scan_result.get(Metric.INVALID_COUNT), 0)
+        self.assertEqual(scan_result.get(Metric.INVALID_PERCENTAGE), 0)
+        self.assertEqual(scan_result.get(Metric.MAX), 5)
+        self.assertEqual(scan_result.get(Metric.MAXS), [5, 4, 3, 2, 1])
+        self.assertEqual(scan_result.get(Metric.MIN), 1)
+        self.assertEqual(scan_result.get(Metric.MINS), [1, 2, 3, 4, 5])
+        self.assertEqual(scan_result.get(Metric.MISSING_COUNT), 1)
+        self.assertEqual(scan_result.get(Metric.MISSING_PERCENTAGE), 8.333333333333334)
+        self.assertAlmostEqual(scan_result.get(Metric.STDDEV), Decimal(1.09544), 4)
+        self.assertEqual(scan_result.get(Metric.SUM), 33)
+        self.assertEqual(scan_result.get(Metric.UNIQUENESS), 40)
+        self.assertEqual(scan_result.get(Metric.UNIQUE_COUNT), 2)
+        self.assertEqual(scan_result.get(Metric.VALID_COUNT), 11)
+        self.assertEqual(scan_result.get(Metric.VALID_PERCENTAGE), 91.66666666666667)
+        self.assertEqual(scan_result.get(Metric.VALUES_COUNT), 11)
+        self.assertEqual(scan_result.get(Metric.VALUES_PERCENTAGE), 91.66666666666667)
+        self.assertAlmostEqual(scan_result.get(Metric.VARIANCE), Decimal(1.2000), 4)
+        self.assertEqual(24, len(scan_result.measurements))

--- a/tests/local/warehouse/validity/test_number_validity_formats.py
+++ b/tests/local/warehouse/validity/test_number_validity_formats.py
@@ -148,7 +148,7 @@ class TestNumberValidityFormats(SqlTestCase):
              f"('-9.99 USD')",
              "('3 sacks of cow dung')",
              "('2 goats and my daughter')"])
-    
+
         scan_result = self.scan({
             KEY_METRICS: [
                 Metric.INVALID_COUNT,
@@ -162,13 +162,14 @@ class TestNumberValidityFormats(SqlTestCase):
                 }
             }
         })
-    
+
+        print([m.metric for m in scan_result.measurements])
         self.assertEqual(scan_result.get(Metric.VALUES_COUNT, 'name'), 10)
         self.assertEqual(scan_result.get(Metric.INVALID_COUNT, 'name'), 2)
         self.assertEqual(scan_result.get(Metric.INVALID_PERCENTAGE, 'name'), 20.0)
         self.assertEqual(scan_result.get(Metric.VALID_COUNT, 'name'), 8)
         self.assertEqual(scan_result.get(Metric.VALID_PERCENTAGE, 'name'), 80.0)
-    
+
     def test_number_money_eur(self):
         self.sql_recreate_table(
             [f"name {self.dialect.data_type_varchar_255}"],
@@ -182,7 +183,7 @@ class TestNumberValidityFormats(SqlTestCase):
              f"('-9,99 EUR')",
              "('1000 Nis')",
              "('10 shrubberies')"])
-    
+
         scan_result = self.scan({
             KEY_METRICS: [
                 Metric.INVALID_COUNT,
@@ -196,13 +197,13 @@ class TestNumberValidityFormats(SqlTestCase):
                 }
             }
         })
-    
+
         self.assertEqual(scan_result.get(Metric.VALUES_COUNT, 'name'), 10)
         self.assertEqual(scan_result.get(Metric.INVALID_COUNT, 'name'), 2)
         self.assertEqual(scan_result.get(Metric.INVALID_PERCENTAGE, 'name'), 20.0)
         self.assertEqual(scan_result.get(Metric.VALID_COUNT, 'name'), 8)
         self.assertEqual(scan_result.get(Metric.VALID_PERCENTAGE, 'name'), 80.0)
-    
+
     def test_number_money_gbp(self):
         self.sql_recreate_table(
             [f"name {self.dialect.data_type_varchar_255}"],
@@ -216,7 +217,7 @@ class TestNumberValidityFormats(SqlTestCase):
              f"('-9.99 GBP')",
              "('1 hamster and 1 basket of elderberries')",
              "('12 million Twitter followers')"])
-    
+
         scan_result = self.scan({
             KEY_METRICS: [
                 Metric.INVALID_COUNT,
@@ -230,13 +231,13 @@ class TestNumberValidityFormats(SqlTestCase):
                 }
             }
         })
-    
+
         self.assertEqual(scan_result.get(Metric.VALUES_COUNT, 'name'), 10)
         self.assertEqual(scan_result.get(Metric.INVALID_COUNT, 'name'), 2)
         self.assertEqual(scan_result.get(Metric.INVALID_PERCENTAGE, 'name'), 20.0)
         self.assertEqual(scan_result.get(Metric.VALID_COUNT, 'name'), 8)
         self.assertEqual(scan_result.get(Metric.VALID_PERCENTAGE, 'name'), 80.0)
-    
+
     def test_number_money_rmb(self):
         self.sql_recreate_table(
             [f"name {self.dialect.data_type_varchar_255}"],
@@ -250,7 +251,7 @@ class TestNumberValidityFormats(SqlTestCase):
              f"('-9.99 RMB')",
              "('10 hugs')",
              "('1.67 x 10^21 water molecules')"])
-    
+
         scan_result = self.scan({
             KEY_METRICS: [
                 Metric.INVALID_COUNT,
@@ -264,13 +265,13 @@ class TestNumberValidityFormats(SqlTestCase):
                 }
             }
         })
-    
+
         self.assertEqual(scan_result.get(Metric.VALUES_COUNT, 'name'), 10)
         self.assertEqual(scan_result.get(Metric.INVALID_COUNT, 'name'), 2)
         self.assertEqual(scan_result.get(Metric.INVALID_PERCENTAGE, 'name'), 20.0)
         self.assertEqual(scan_result.get(Metric.VALID_COUNT, 'name'), 8)
         self.assertEqual(scan_result.get(Metric.VALID_PERCENTAGE, 'name'), 80.0)
-    
+
     def test_number_money_chf(self):
         self.sql_recreate_table(
             [f"name {self.dialect.data_type_varchar_255}"],
@@ -284,7 +285,7 @@ class TestNumberValidityFormats(SqlTestCase):
              self.qualify_string(f"('-9.99 CHF')"),
              "('1 Chuck Norris roundhouse kick')",
              "('16 candles')"])
-    
+
         scan_result = self.scan({
             KEY_METRICS: [
                 Metric.INVALID_COUNT,
@@ -298,13 +299,13 @@ class TestNumberValidityFormats(SqlTestCase):
                 }
             }
         })
-    
+
         self.assertEqual(scan_result.get(Metric.VALUES_COUNT, 'name'), 10)
         self.assertEqual(scan_result.get(Metric.INVALID_COUNT, 'name'), 2)
         self.assertEqual(scan_result.get(Metric.INVALID_PERCENTAGE, 'name'), 20.0)
         self.assertEqual(scan_result.get(Metric.VALID_COUNT, 'name'), 8)
         self.assertEqual(scan_result.get(Metric.VALID_PERCENTAGE, 'name'), 80.0)
-    
+
     def test_number_money(self):
         self.sql_recreate_table(
             [f"name {self.dialect.data_type_varchar_255}"],
@@ -324,7 +325,7 @@ class TestNumberValidityFormats(SqlTestCase):
              f"('-99.99 CHF')",
              "('2 garbage bags full of old hair and nails')",
              "('7 samurais')"])
-    
+
         scan_result = self.scan({
             KEY_METRICS: [
                 Metric.INVALID_COUNT,
@@ -338,7 +339,7 @@ class TestNumberValidityFormats(SqlTestCase):
                 }
             }
         })
-    
+
         self.assertEqual(scan_result.get(Metric.VALUES_COUNT, 'name'), 16)
         self.assertEqual(scan_result.get(Metric.INVALID_COUNT, 'name'), 2)
         self.assertEqual(scan_result.get(Metric.INVALID_PERCENTAGE, 'name'), 12.5)

